pkg/irc/client.go

```diff
@@ -0,0 +1,252 @@
+package irc
+
+import (
+	"bufio"
+	"crypto/tls"
+	"io"
+	"net"
+	"net/textproto"
+	"strings"
+	"sync"
+	"time"
+)
+
+var (
+	// ReadBuffer assigns a buffer size to the read channel
+	ReadBuffer = 32
+	// WriteBuffer assigns a buffer size to the write channel
+	WriteBuffer = 0
+	// Address sets the address that the client will connect to
+	Address = "irc.chat.twitch.tv:6667"
+	// AddressTLS sets the address that the client will connect to in TLS mode
+	AddressTLS = "irc.chat.twitch.tv:6697"
+)
+
+// Client handles the IRC connection and incoming & outgoing messages.
+// The client requires you to respond to PING messages manually
+// as well as keep track of which channels you're connected to using the incoming JOIN & PART messages
+type Client struct {
+	user  string
+	oauth string
+
+	capabilities []string
+
+	// UseTLS determines whether the IRC connects with or without TLS, needs to be set before you call Connect, default = true
+	UseTLS bool
+
+	read  chan string
+	write chan []byte
+
+	serverDisconnect closer
+	clientDisconnect closer
+
+	onConnect func()
+	onMessage func(msg *Message, err error)
+}
+
+// New returns a new client
+func New(user, oauth string) *Client {
+	return &Client{
+		user:   user,
+		oauth:  oauth,
+		UseTLS: true,
+		read:   make(chan string, ReadBuffer),
+		write:  make(chan []byte, WriteBuffer),
+	}
+}
+
+// NewAnon returns an anonymous client, useful for testing, or small read-only bots
+func NewAnon() *Client {
+	return &Client{
+		user:   "justinfan4321",
```

Comment:
extremely important suggestion ahead
```suggestion
		user:   "justinfan7777",
```

Reply:
Apparently it's recommended to use 5 numbers, so it's now 77777